salt/modules/saltutil.py

```diff
@@ -1444,6 +1445,14 @@ def _exec(client, tgt, fun, arg, timeout, tgt_type, ret, kwarg, **kwargs):
             'ret': ret, 'cli': True, 'kwarg': kwarg, 'sub': kwargs['subset'],
         }
         del kwargs['subset']
+    elif kwargs.get('asynchronous'):
+        cmd_kwargs = {
+            'tgt': tgt, 'fun': fun, 'arg': arg, 'tgt_type': tgt_type,
+            'ret': ret, 'kwarg': kwarg
+        }
+        del kwargs['asynchronous']
+        # run_job doesnt need processing like the others
+        return client.run_job(**cmd_kwargs)
```

Comment:
Passed kwargs aren't ever applied to cmd_kwargs, unlike the other cases. Keeping consistent behavior would be good.

A thing I'm not clear on - why does this chunk bypass calling `_cmd` like the other ways the function is called?